src/pkg/clouds/gcp/project.go

```diff
@@ -160,7 +160,7 @@ func (gcp Gcp) EnsureProjectExists(ctx context.Context, projectName string) (*re
 	return project, nil
 }
 
-func (gcp Gcp) GetProjectNumber(ctx context.Context) (int64, error) {
+func (gcp *Gcp) GetProjectNumber(ctx context.Context) (int64, error) {
```

Comment:
Good catch, but I think the bigger problem is probably the gcp object should not have contained the ID property (my fault), and thus make the gcp object stateless.
I would suggest remove the ProjectNumber from the struct.